topostats/processing.py

```diff
@@ -430,25 +398,23 @@ def run_disordered_tracing(
         disordered_trace_grainstats = pd.DataFrame()
         disordered_tracing_stats_image = pd.DataFrame()
         try:
-            # run image using directional grain masks
-            for direction, _ in grain_masks.items():
-                # Check if there are grains
-                assert len(grain_masks[direction].shape) == 3, "Grain masks should be 3D tensors"
-                dna_class_mask = grain_masks[direction][:, :, 1]
-                if np.max(dna_class_mask) == 0:
+
+            grain_crop_direction: GrainCropsDirection
+            for direction, grain_crop_direction in image_grain_crops.__dict__.items():
+
+                if grain_crop_direction is None:
                     LOGGER.warning(
                         f"[{filename}] : No grains exist for the {direction} direction. Skipping disordered_tracing for {direction}."
                     )
                     raise ValueError(f"No grains exist for the {direction} direction")
```

Comment:
Because of the presence of .above and .below every time regardless of the chosen option, the above section will always iterate through both, thus ValueError needs to be removed as we don't want to break this loop.

Then we can use:
```
if grain_crop_direction is None:
	...
else:
	trace_image_disordered(...
```

Reply:
So the suggestion is simply to remove the `raise` ?

```suggestion
                    raise ValueError(f"No grains exist for the {direction} direction")
```

Reply:
Yes but you'll also want to avoid actually trying to run the disordered tracing with an empty dict as that's not handled well I think so it needs to be in the else clause